components/kyma-environment-broker/cmd/broker/main.go

```diff
@@ -673,24 +673,28 @@ func NewProvisioningProcessingQueue(ctx context.Context, provisionManager *proce
 			step:      provisioning.NewBTPOperatorOverridesStep(db.Operations()),
 		},
 		{
-			condition: skipForOwnClusterPlan,
+			condition: provisioning.SkipForOwnClusterPlan,
 			stage:     createRuntimeStageName,
 			step:      provisioning.NewCreateRuntimeWithoutKymaStep(db.Operations(), db.RuntimeStates(), db.Instances(), provisionerClient),
 		},
 		{
-			condition: doForOwnClusterPlanOnly,
+			condition: provisioning.DoForOwnClusterPlanOnly,
 			stage:     createRuntimeStageName,
 			step:      provisioning.NewCreateRuntimeForOwnClusterStep(db.Operations(), db.Instances()),
 		},
-		// check the runtime status
 		{
 			stage:     createRuntimeStageName,
 			step:      provisioning.NewCheckRuntimeStep(db.Operations(), provisionerClient, cfg.Provisioner.ProvisioningTimeout),
-			condition: skipForOwnClusterPlan,
+			condition: provisioning.SkipForOwnClusterPlan,
 		},
 		{
 			stage: createRuntimeStageName,
-			step:  provisioning.NewGetKubeconfigStep(db.Operations(), provisionerClient),
+			step:  provisioning.NewGetKubeconfigStep(db.Operations(), provisionerClient, k8sClientProvider),
+		},
+		{
+			condition: provisioning.WhenBTPOperatorCredentialsProvided,
```

Comment:
do we want KEB to do this for all new SKRs and only for provisioning? Theoretically, BTP Operator credentials can be provided during instance update (`PATCH` for `/service_instances/{ID}` endpoint) and this way KEB will behave differently for update and provision.

Reply:
Yes. This was agreed upon during our discussions with @PK85.

Reply:
During 'PATCH' call we should update/recreate/overwrite the secret when the step is added there. We should consider adding this step for update operation.